src/Microsoft.DocAsCode.Build.Common/Reference/ApplyOverwriteDocument.cs

```diff
@@ -94,8 +94,12 @@ select new
                 {
                     pair.model.LinkToUids = pair.model.LinkToUids.Union(overwriteDocumentModel.LinkToUids);
                     pair.model.LinkToFiles = pair.model.LinkToFiles.Union(overwriteDocumentModel.LinkToFiles);
-                    pair.model.FileLinkSources = pair.model.FileLinkSources.Union(overwriteDocumentModel.FileLinkSources.ToImmutableDictionary(k => k.Key, k => k.Value.ToImmutableList())).ToImmutableDictionary();
-                    pair.model.UidLinkSources = pair.model.UidLinkSources.Union(overwriteDocumentModel.UidLinkSources.ToImmutableDictionary(k => k.Key, k => k.Value.ToImmutableList())).ToImmutableDictionary();
+                    pair.model.FileLinkSources = pair.model.FileLinkSources.Union(
```

Comment:
I just added new method to suitable for IImmutableDictionary

---
In reply to: [100939213](https://github.com/dotnet/docfx/pull/1298#discussion_r100939213) [](ancestors = 100939213)